### PR TITLE
Refresh state relations

### PR DESCRIPTION
**Pull request by slemrmartin**
TPINVTRY-837

---

Items are collected in `sweep scopes` (for mark & sweep algorithm) and each model's data in payload are recorded to `refresh_state_parts` table. 

It means that saved items can be paired with specific payload.

### CODE REVIEW

**Comment by slemrmartin** [comment]
Based on #182.

@agrare I added only "last refresh state" and not a "last successful refresh state", because refresh_state_part ID can be added to inventory objects **before** saving. 

Save successful refresh state means resaving all data again which would slow down the whole process

**Comment by agrare** [comment]
Looks good @slemrmartin I don't see a last_refresh_state change though did you forget to push something or is that a followup?

**Comment by slemrmartin** [comment]
ah, right, I forget to push `inventory_refresh` PR

**Comment by slemrmartin** [comment]
@agrare this can be merged without inventory refresh's PR. 
I'm not sure if `"inventory_refresh", "~> 0.3.2"` in gemspec is enough, or it should be updated.

I meanif it's possible that builder of image for openshift could use cache and doesn't update dependency..?

**Comment by slemrmartin** [comment]
@agrare my comments ^^ was wrong. Persister changes must be implemented to core's persister.

So it's ready

**Comment by slemrmartin** [comment]
* depends on https://github.com/RedHatInsights/topological_inventory-core/pull/184

File: lib/topological_inventory/schema/base.rb [hunk 1]
```diff
@@ -21,6 +26,20 @@ def self.klass_for(name)
         # checks whether it loads what we want
         klass if klass.to_s == klass_name
       end
+
+      private
+
+      def link_data_to_refresh_state_part
```
Comment (agrare) [inline lib/topological_inventory/schema/base.rb]: @slemrmartin I wonder if this makes more sense in the topological_inventory-persister?  Thinking maybe in this [method](https://github.com/RedHatInsights/topological_inventory-persister/blob/master/lib/topological_inventory/persister/workflow.rb#L87-L98)

**Pull request merged**